tezos/baking/src/helpers.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use storage::num_from_slice;
+use crypto::blake2b;
+
+macro_rules! merge_slices {
+    ( $($x:expr),* ) => {{
+        let mut res = vec![];
+        $(
+            res.extend_from_slice($x);
+        )*
+        res
+    }}
+}
+
+// cycle in which is given level
+// level 0 (genesis block) is not part of any cycle (cycle 0 starts at level 1), hence the -1
+pub fn cycle_from_level(level: i32, blocks_per_cycle: i32) -> i32 {
+    (level - 1) / blocks_per_cycle
+}
+
+// the position of the block in its cycle
+// level 0 (genesis block) is not part of any cycle (cycle 0 starts at level 1)
+// hence the blocks_per_cycle - 1 for last cycle block
+pub fn level_position(level:i32, blocks_per_cycle:i32) -> i32 {
+    let cycle_position = (level % blocks_per_cycle) - 1;
+    if cycle_position < 0 { //for last block
+        blocks_per_cycle - 1
+    } else {
+        cycle_position
+    }
+}
+
+// tezos PRNG
+pub fn get_random_number(state: Vec<u8>, nonce_size: usize, blocks_per_cycle: i32, use_string_bytes: &[u8], level: i32, offset: i32, bound: i32) -> Result<(i32, Vec<u8>), failure::Error> {
```

Comment:
- document this function
- it's not clear what `(i32, Vec<u8>)` returned by this function represents. Consider using type alias or create a new type.
- try to avoid using generic `failure::Error` on public functions